fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1640,9 +1690,188 @@ public String getSchemaForRequestContext(Connection connection) throws FHIRPersi
+    
+                    // Use an OperationOutcome Issue to let the caller know that some work was performed
+                    final String diag = "Processed " + rir.getResourceType() + "/" + rir.getLogicalId();
+                    operationOutcomeResult.issue(Issue.builder().code(IssueType.INFORMATIONAL).severity(IssueSeverity.INFORMATION).diagnostics(com.ibm.fhir.model.type.String.of(diag)).build());
+                } else {
+                    // Reasonable to assume that this resource was deleted because we can't read it
+                    final String diag = "Failed to read resource: " + rir.getResourceType() + "/" + rir.getLogicalId();
+                    operationOutcomeResult.issue(Issue.builder().code(IssueType.NOT_FOUND).severity(IssueSeverity.WARNING).diagnostics(com.ibm.fhir.model.type.String.of(diag)).build());
+                }                
+            }
+            
+        } catch(FHIRPersistenceFKVException e) {
+            getTransaction().setRollbackOnly();
+            throw e;
+        } catch(FHIRPersistenceException e) {
+            getTransaction().setRollbackOnly();
+            throw e;
+        } catch (DataAccessException dax) {
+            getTransaction().setRollbackOnly();
+            
+            // It's possible this is a deadlock exception, in which case it could be considered retryable
+            if (dax.isTransactionRetryable()) {
+                log.log(Level.SEVERE, "retryable error", dax);
+                FHIRPersistenceDataAccessException fpx = new FHIRPersistenceDataAccessException("Data access error while performing a reindex operation.");
+                fpx.setTransactionRetryable(true);
+                throw fpx;
+            } else {
+                log.log(Level.SEVERE, "non-retryable error", dax);
+                throw new FHIRPersistenceDataAccessException("Data access error while performing a reindex operation.");
+            }
+        } catch(Throwable e) {
+            getTransaction().setRollbackOnly();
+            // don't chain the exception to avoid leaking secrets
+            FHIRPersistenceException fx = new FHIRPersistenceException("Unexpected error while performing a reindex operation.");
+            log.log(Level.SEVERE, fx.getMessage(), e);
+            throw fx;
+        }
+        finally {
+            log.exiting(CLASSNAME, METHODNAME);
+        }
+        
+        return result;
+    }
     
     @Override
     public String generateResourceId() {
         return logicalIdentityProvider.createNewIdentityValue();
     }
+
+    /**
+     * Each datasource involved in a transaction gets its own TransactionData impl object
+     * which is used to hold parameter data to be inserted into the datbase just prior to
+     * commit.
+     * @param datasourceId
+     * @return the ParameterTransactionDataImpl used to hold onto data to persist at the end of the current transaction
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
What's unchecked here?

Reply:
this.transactionDataImpl = (TransactionDataImpl<ParameterTransactionDataImpl>)tdi;